libraries/cyclestreets-view/src/main/java/net/cyclestreets/views/overlay/POIOverlay.kt

```diff
@@ -45,15 +39,15 @@ class POIOverlay(mapView: CycleMapView) : LiveItemOverlay<POIOverlayItem?>(mapVi
     private val context: Context = mapView.context
     private val activeCategories: MutableList<POICategory> = ArrayList()
     private val overlays: OverlayHelper = OverlayHelper(mapView)
-    private val curScreenCoords = Point()
 
     private val poiIcon = materialIcon(context, GoogleMaterial.Icon.gmd_place)
 
-    private var activeItem: POIOverlayItem? = null
     private var lastFix: IGeoPoint? = null
-    private var bubble: Rect? = null
     private var chooserShowing: Boolean = false
-    private val tapHereText = context.getString(R.string.tap_here)
+
+    init {
+        Bubble.initialise(context, mapView)
+    }
```

Comment:
Based on `Bubble` being a shared singleton, can you confirm you've done testing which plays around with circular route POIs, and you've then cleared the route and played around with POIs normally, and back again, without issues?

Reply:
I always worry I've missed something, so it's good to have your more experienced eye on it.
As I remember, I did extensive testing around the 2 sets of POIs co-existing. (Had some issues which I resolved, but not to do with the Bubble singleton.)
I've added some of these into the test plan attached.  Could you have a look at items 25-38 in the test plan to see if this covers what you had in mind?

Reply:
There is one slight issue in that where the same POI is displayed as both a normal one and a Circular Route one, the bubble displayed will be the Circular Route bubble, which can have slightly less info than the normal POI bubble.

I tried to reverse this, but it was a bit more work than I originally thought, so I left it for now.  Would it be OK to raise this as a separate issue? (@mvl22, thoughts welcome?)

(I wasn't able to do any clever exclusion of the Circular Route ones from the Normal ones, or vice versa, as I didn't have a unique identifier and it would have been too hit and miss to do it by location, name, etc)

Reply:
seems OK to me.